evm/vm/base.py

```diff
@@ -57,9 +72,17 @@ def configure(cls,
                 )
         return type(name, (cls,), overrides)
 
+    def set_stateless(self, is_stateless):
```

Comment:
Hrm, is it possible for us to avoid needing this setter and to require statelessness be declared on instantiation?